homeassistant/components/modbus/base_platform.py

```diff
@@ -218,7 +218,12 @@ def unpack_structure_result(self, registers: list[int]) -> str | None:
                 if isinstance(v_temp, int) and self._precision == 0:
                     v_result.append(str(v_temp))
                 else:
-                    v_result.append(f"{float(v_temp):.{self._precision}f}")
+                    # NaN float detection replace with None
+                    # Issue: https://github.com/home-assistant/core/issues/93297
+                    if v_temp != v_temp:
```

Comment:
This looks wrong. this can never be true.

Reply:
floats that equal NaN are never equal to itself. The array temp variable, "v_temp" and further down the "val_result" are floats. I will provide testing evidence showing the case when NaN is received. 

https://stackoverflow.com/questions/10034149/why-is-nan-not-equal-to-nan

Reply:
you test if the variable if different from the same variable.

If you look at the failing tests, pylint also mark this as an error.

Reply:
Correct, that's one way of detecting NaN. See if it is equal to its self. 
https://stackoverflow.com/questions/10034149/why-is-nan-not-equal-to-nan